src/shell/gmenu.js

```diff
@@ -92,12 +93,19 @@ export class ListBox extends PopupMenu.PopupMenuSection {
         this.actor.add_child(this.box);
 
         // Submenu Container
-        this.sub = new St.BoxLayout({
-            clip_to_allocation: true,
-            vertical: false,
-            visible: false,
-            x_expand: true,
-        });
+        this.sub = Number(PACKAGE_VERSION.split('.')[0]) >= 48
```

Comment:
Part of me wants to create a "feature-test" constant for this, instead of having to do the string-processing and comparison every time. Something like,

```js
// in src/shell/utils.js maybe
export const HAVE_ST_ORIENTATION = new Boolean(
    Number(PACKAGE_VERSION.split('.')[0]) >= 48);

// Everywhere else
import {HAVE_ST_ORIENTATION} from '../shell/utils.js';
this.sub = new St.BoxLayout(
    HAVE_ST_ORIENTATION
    ? { ... }
    : { ... }
);
```

Reply:
(If it does go in `src/shell/utils.js`, the `PACKAGE_VERSION` import and the definition of `HAVE_ST_ORIENTATION` will need the same conditional logic as the `St` import, to protect `prefs.js`'s use pf that mpdule.)

Reply:
@ferdnyc the latest 2 commits should be of interest, let me know what you think

I added feature flags in the first one, and the second one eliminates dynamic imports (which started to feel really icky) by moving the setup code into the `utils` folder, which makes much more sense to me, since it's common shared code